From axboe@kernel.dk Mon Jan 30 05:47:34 2006
Date: Sun, 29 Jan 2006 20:57:33 +0100
From: Jens Axboe <axboe@suse.de>
To: James Bottomley <James.Bottomley@SteelEye.com>
Cc: Pasi =?iso-8859-1?Q?K=E4rkk=E4inen?= <pasik@iki.fi>,
	Nix <nix@esperi.org.uk>, Ariel <askernel2615@dsgml.com>,
	Jamie Heilman <jamie@audible.transient.net>,
	Chase Venters <chase.venters@clientec.com>,
	Arjan van de Ven <arjan@infradead.org>,
	linux-ide@vger.kernel.org, linux-kernel@vger.kernel.org,
	linux-scsi@vger.kernel.org
Subject: SCSI: turn off ordered flush barriers
Message-ID: <20060129195733.GH13831@suse.de>
Content-Disposition: inline

Turn off ordered flush barriers for SCSI driver, since the SCSI barrier
code has a command leak.

Signed-off-by: Jens Axboe <axboe@suse.de>
Signed-off-by: Chris Wright <chrisw@sous-sol.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---

 drivers/scsi/scsi_lib.c |    5 -----
 1 files changed, 5 deletions(-)

Index: linux-2.6.15.3/drivers/scsi/scsi_lib.c
===================================================================
--- linux-2.6.15.3.orig/drivers/scsi/scsi_lib.c
+++ linux-2.6.15.3/drivers/scsi/scsi_lib.c
@@ -1534,11 +1534,6 @@ struct request_queue *scsi_alloc_queue(s
 	 */
 	if (shost->ordered_tag)
 		blk_queue_ordered(q, QUEUE_ORDERED_TAG);
-	else if (shost->ordered_flush) {
-		blk_queue_ordered(q, QUEUE_ORDERED_FLUSH);
-		q->prepare_flush_fn = scsi_prepare_flush_fn;
-		q->end_flush_fn = scsi_end_flush_fn;
-	}
 
 	if (!shost->use_clustering)
 		clear_bit(QUEUE_FLAG_CLUSTER, &q->queue_flags);
